Reseed errors out with numpy legacy seeding

### Python Version

3.13

### pytest Version

8.4.2

### Package Version

4.0.0

### Description

When paired with `numpy` `v2.3.2` using macOS arm64, I get the following error when `--randomly-seed` is not provided:

```
    def pytest_runtest_teardown(item: Item) -> None:
        if item.config.getoption("randomly_reset_seed"):
>           _reseed(item.config, _crc32(item.nodeid) + 1)

.venv/lib/python3.13/site-packages/pytest_randomly/__init__.py:212: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
.venv/lib/python3.13/site-packages/pytest_randomly/__init__.py:180: in _reseed
    reseed(seed)
.venv/lib/python3.13/site-packages/thinc/util.py:96: in fix_random_seed
    numpy.random.seed(seed)
numpy/random/mtrand.pyx:4828: in numpy.random.mtrand.seed
    ???
numpy/random/mtrand.pyx:254: in numpy.random.mtrand.RandomState.seed
    ???
numpy/random/_mt19937.pyx:168: in numpy.random._mt19937.MT19937._legacy_seeding
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   ValueError: Seed must be between 0 and 2**32 - 1

numpy/random/_mt19937.pyx:182: ValueError
```

--- a/thinc/util.py
+++ b/thinc/util.py
@@ -95,10 +95,16 @@ def fix_random_seed(seed: int = 0) -> None:  # pragma: no cover
     random.seed(seed)
     numpy.random.seed(seed % 2**32)
     if has_torch:
-        torch.manual_seed(seed)
+        # Must be within the inclusive range [-0x8000_0000_0000_0000, 0xffff_ffff_ffff_ffff]
+        # https://github.com/pytorch/pytorch/blob/d38164a545b4a4e4e0cf73ce67173f70574890b6/torch/random.py#L32-L41
+        torch.manual_seed(seed % 0xffff_ffff_ffff_ffff)
     if has_cupy_gpu:
+        # cupy has no documented range limit on seed as of 2026-01-15
+        # https://docs.cupy.dev/en/latest/reference/generated/cupy.random.seed.html
         cupy.random.seed(seed)
         if has_torch and has_torch_cuda_gpu:
+            # torch.cuda has no documented range limit on seed as of 2026-01-15
+            # https://docs.pytorch.org/docs/stable/generated/torch.cuda.manual_seed_all.html
             torch.cuda.manual_seed_all(seed)
             torch.backends.cudnn.deterministic = True
             torch.backends.cudnn.benchmark = False

--- a/thinc/util.py
+++ b/thinc/util.py
@@ -93,7 +93,7 @@ def gpu_is_available():
 def fix_random_seed(seed: int = 0) -> None:  # pragma: no cover
     """Set the random seed across random, numpy.random and cupy.random."""
     random.seed(seed)
-    numpy.random.seed(seed)
+    numpy.random.seed(seed % 2**32)
     if has_torch:
         torch.manual_seed(seed)
     if has_cupy_gpu:

